Fix(pager): Resolve background color before staring pager

zemzale opened the pull request:
**Description**
So far we have started the pager and then got background color glamour
style from enviorment.

If enviorment value is not set it defaults to auto. Which in turn uses
termenv to resolve background color using some terminal escape code
sequences. The problem with this is that doing this when a pager like
less is running, just doesn't work and leaves less in a broken state.

To fix this we are now checking the background color manually and before
we start the pager.

**Related Issue**
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
Resolves #636

**How Has This Been Tested?**
Fixes hangs for less on my machine.

**Types of changes**
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation
- [ ] Chore (Related to CI or Packaging to platforms)

zemzale committed:
fix(utils): Resolve bgColor before pager starts

So far we have started the pager and then got background color glamour
style from enviorment.

If enviorment value is not set it defaults to auto. Which in turn uses
termenv to resolve background color using some terminal escape code
sequences. The problem with this is that doing this when a pager like
less is running, just doesn't work and leaves less in a broken state.

To fix this we are now checking the background color manually and before
we start the pager.

Issue #636

profclems commented:
@zemzale kindly rebase the branch

zemzale commented:
@profclems Will do. There seem to be some issues about this with tests anyways. So gotta fix that first

zemzale commented:
Rebased and tests working

profclems commented on pkg/iostreams/iostreams.go:
```diff
@@ -166,6 +169,31 @@ func (s *IOStreams) IsOutputTTY() bool {
 	return s.IsErrTTY && s.IsaTTY
 }
 
+func (s *IOStreams) ResolveBackgroundColor() string {
+	style := os.Getenv("GLAMOUR_STYLE")
```
```suggestion
func (s *IOStreams) ResolveBackgroundColor(style string) string {
        if(style == "") {
		style = os.Getenv("GLAMOUR_STYLE")
        }
```

profclems commented on commands/issue/view/issue_view.go:
```diff
@@ -93,6 +93,7 @@ func NewCmdView(f *cmdutils.Factory) *cobra.Command {
 				}
 			}
 
+			f.IO.ResolveBackgroundColor()
```
```suggestion
			f.IO.ResolveBackgroundColor(opts.GlamourStyle)
```

profclems commented on commands/mr/view/mr_view.go:
```diff
@@ -77,6 +77,7 @@ func NewCmdView(f *cmdutils.Factory) *cobra.Command {
 				}
 			}
 
+			f.IO.ResolveBackgroundColor()
```
```suggestion
			f.IO.ResolveBackgroundColor(glamourStyle)
```

profclems reviewed (changes requested):
Thanks for working on this.

I have added a few suggestions in the inline review comments: we allow users set the `glamour_style` in the config so we should consider that. 
So in the `mr_view` and `issue_view`, we get the `glamour_style` from the config and pass to the `ResolveBackgroundColor(style string)`.

zemzale commented:
Fixed the issue with not config-defined styles. 

I also removed the `GlamourStyle` variables being passed around since we really don't need that anymore, because we are resolving the style before, and it's available from IOStreams.

profclems reviewed (approved):
LGTM
